packages/angular/src/tracing.ts

```diff
@@ -246,13 +244,19 @@ export class TraceDirective implements OnInit, AfterViewInit {
 
   private _tracingSpan?: Span;
 
+  public constructor(private readonly _host: ElementRef<HTMLElement>) {}
+
   /**
    * Implementation of OnInit lifecycle method
    * @inheritdoc
    */
   public ngOnInit(): void {
     if (!this.componentName) {
-      this.componentName = UNKNOWN_COMPONENT;
+      // Technically, the `trace` binding should always be provided.
+      // However, if it is incorrectly declared on the element without a
+      // value (e.g., `<app-component trace />`), we fall back to using `tagName`
+      // (which is `APP-COMPONENT`).
+      this.componentName = this._host.nativeElement.tagName.toLowerCase();
```

Comment:
Are we sure this chain is always defined? Across all Angular versions (14-19)? Otherwise, let's maybe change this to 
```suggestion
      this.componentName = this._host?.nativeElement?.tagName?.toLowerCase() || ''unknown';
```

just to be safe

Reply:
No, this is defined in all Angular versions.